src/main/java/io/reactivex/internal/operators/flowable/FlowableCache.java

```diff
@@ -364,15 +370,12 @@ public void replay() {
                         }
                     }
 
-                    if (valuesProduced != 0) {
-                        BackpressureHelper.producedCancel(rq, valuesProduced);
```

Comment:
Having a separate emitted, there is no need to decrement the requested amount, saving on an atomic instruction.